src/librustc_typeck/structured_errors.rs

```diff
@@ -50,8 +50,7 @@ impl<'tcx> StructuredDiagnostic<'tcx> for VariadicError<'tcx> {
     }
 
     fn code(&self) -> DiagnosticId {
-        syntax::diagnostic_used!(E0617);
-        DiagnosticId::Error("E0617".to_owned())
+        syntax::stringify_error_code!(E0617)
```

Comment:
If we're touching most of these locations, would it be worth also moving to (loosely speaking) `rustc_errors::error_code!(...)`? Maybe out of scope for this PR. I'm also not a fan of the `stringify_` naming, as that's an implementation detail IMO.

Reply:
Renamed. :)